### ISSUE TITLE
LoRa Communication Setup

### ISSUE THREAD

**Issue by David-Shang** [bellabaohaha/CITS5206-Gp2#36]
Set up and test the LoRa communication between the Soil Moisture Monitoring System and the LoRa Gateway.

**Comment by asadmaza** [comment]
Hey @ITwoaini  Hanni,

To clarify, your task here is to setup the LoRa communication between the soil moisture monitoring node and the Internet/LoRa Gateway (TTGO). You will have to work with Uday/Dharani on this one to ensure they have their sender setup and you have your receiver setup (see LilyGo library that I linked under components list I created).

I imagine it will be pretty straight forward to send from their TTGO (Soil Moisture Node) to yours (setup as LoRa Receiver). Once you achieve that, I would then work with David as he's responsible for Data transmission to AWS IoT Core. If you have any confusion, it's best you come in to uni tomorrow (05/09) as I will be available to guide you all towards a viable solution.

What I explained about the internet gateway and adding the metadata as per the dataflow diagram and connecting and transmitting the data to IoT Core is actually David's task - please check/remind him when he's available. @David-Shang

**Comment by ITwoaini** [comment]
Hey Asad, thanks for the explanation. I'll come to the university tomorrow (05/09) to discuss.

**Comment by ITwoaini** [comment]
Communication setup:

We followed Australia’s frequency regulations, configuring the system to use a 915 MHz frequency, a 125 kHz signal bandwidth, and a spreading factor of 10. While these parameters are theoretically expected to provide a good balance between energy efficiency and communication range, initial tests indicated that the system did not perform optimally. Even at distances less than 100 meters, we experienced communication issues, suggesting that further optimization is necessary for reliable operation in large-scale agricultural scenarios. While these parameters are not exhaustive, they represent the key settings optimized for our project. Further tuning of additional parameters can further improve communication reliability. We also use a sync word of 0xAB to ensure the sensor board and gateway communicate properly, while avoiding interference from other LoRa networks.

**Comment by ITwoaini** [comment]
**Test data and analysis for communication:**

We conducted a brief test in a controlled indoor laboratory environment to verify whether the LoRa communication between the sensor board and the gateway was functioning properly. Using two TTGO T-Beam development boards—one as the sender and the other as the receiver—the sender transmitted data packets every 2 seconds containing three simulated soil moisture sensor readings (values: 123, 321, 456).

The receiver continuously monitored the LoRa channel. While the receiver successfully received most of the transmitted data packets, we observed some data corruption and packet loss. Out of 20 data packets sent, 16 were correctly received, resulting in a packet loss rate of 20%.

_Below table shows the first ten rows of Test Results of Lab Testing_

Packet No.  Mock Sensor Data	RSSI (dBm)	SNR (dB)
1	[123,321,456]	                         -21	                 10.00
2	[123,321,456]	                         -21	                 10.50
3	[123,321,456]	                         -21	                 10.25
4	[123,321,456]	                         -21	                 10.25
5	[123,321,456]	                         -21	                 10.25
6	[123,321,456]	                         -21	                  9.75
7	[Data Corrupted]	                 -21	                  —
8	[123,321,456]	                         -21	                 10.25
9	[123,321,456]	                         -21	                 10.25
10	[Data Corrupted]	                 -21	                  —



The RSSI values remained stable at around -21 dBm, and the SNR values ranged from 9.75 dB to 10.50 dB. According to Semtech's guidelines[1], reliable communication typically requires RSSI values higher than -120 dBm and SNR values above 0 dB. Our observed values fall well within these ideal ranges, indicating good signal strength and quality in the laboratory environment at short range.

Despite the ideal RSSI and SNR values, we encountered instances of packet loss and data corruption. This suggests that factors other than signal strength and quality are affecting communication reliability. Given the controlled environment, potential causes may include interference from nearby wireless devices (such as Wi-Fi networks) or timing issues in the LoRa protocol. Additionally, the lack of an acknowledgment mechanism in our current communication setup may have contributed to data loss, as there is no retransmission when packets are corrupted or lost.


[1] _Semtech Corporation, "AN1200.22 LoRa Modulation Basics," Application Note, May 2015._
